pkg/trace/metrics/client.go

```diff
@@ -20,12 +21,23 @@ type StatsClient interface {
 	Flush() error
 }
 
+var m sync.Mutex
+
+// SetClient sets the global Statsd client.
+func SetClient(client StatsClient) {
+	m.Lock()
+	defer m.Unlock()
+	Client = client
+}
+
```

Comment:
An atomic would be more efficient, since (i think) we're just protecting the `Client` reference itself.

Reply:
Do you mean [atomic.Value](https://pkg.go.dev/sync/atomic#Value) ?